### PR TITLE
Add "stress" e2e test

### PR DESCRIPTION
**Pull request by andrewgordstewart**
This adds a test file setting up
- one file
- one seeders
- three leechers

There is a more complex, slightly stale test that is skipped. It should be updated to ensure that each actor signs with its own private key

Some other changes were made along the way trying to troubleshoot these tests:
- mostly adding logs
- the simple-hub does not crash on failed deposits

### CODE REVIEW

**Comment by andrewgordstewart** [comment]
(Ignore this close: testing a github projects feature)

**Comment by snario** [comment]
What's the status of this PR?

File: packages/e2e-tests/puppeteer/__tests__/web3torrent/stress.test.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    A = 'A',
+    B = 'B',
+    C = 'C',
+    D = 'D'
+  }
+
+  const leechers: Label[] = [Label.B, Label.C, Label.D];
+  const labels: Label[] = leechers.concat([Label.A]);
+
+  type T = {browser: Browser; metamask: Dappeteer; tab: Page};
```
Comment (lalexgap) [inline packages/e2e-tests/puppeteer/__tests__/web3torrent/stress.test.ts:42]: Nit: could this be a more descriptive type name?

File: packages/web3torrent/src/clients/payment-channel-client.ts [hunk 2]
```diff
@@ -170,12 +170,15 @@ export class PaymentChannelClient {
   }
 
   insertIntoChannelCache(channelState: ChannelState) {
-    this.channelCache = {...this.channelCache, [channelState.channelId]: channelState};
+    // TODO: The joinChannel promise never resolves, meaning this function is not called
```
Comment (lalexgap) [inline packages/web3torrent/src/clients/payment-channel-client.ts:173]: Is this still the case? I thought we fixed the `joinChannel` not resolving but I could be wrong.

Comment (andrewgordstewart) [reply packages/web3torrent/src/clients/payment-channel-client.ts]: It's an out of date comment, thanks to a three-way merge

**Pull request merged**